### DISCUSSION TITLE
When doing web uploads files get stuck on "upload complete, waiting" even if succeeded

### DISCUSSION THREAD

**Discussion by Fogest** [jonaswinkler/paperless-ng#1459] [Support]
My files are getting stuck in the status seen in the image below:

![image](https://user-images.githubusercontent.com/1163628/143980987-3700f901-ce70-4ad1-be14-eecb68ae90a8.png)

I have migrated from paperless-ng I had setup on a previous server to a new one. And everything but the web upload based consumption seems to be working. On my old installation there was no issue with web based uploads. Now I do not understand the what the issue is here. According to the logs (seen in image below), the file is processed without issue. And if I check the website the document does indeed get uploaded properly with OCR and everything. If the file has a different error such as it being a duplicate this still does the same and is stuck with the same message. However again the logs will properly indicate the error. It's almost as if the updated status of the file upload isn't getting back to the browser for some reason.

Did I mess something up with the configuration? 

![image](https://user-images.githubusercontent.com/1163628/143980970-a9d0c45c-e0d7-458a-8d5e-814e26f844db.png)

I am using the docker setup with an unchanged docker compose YML file, and only a few basic variables set in the docker compose env file. Nothing different from my old setup there.

**Comment by siancu** [comment]
I am seeing exactly the same behavior. Where do you run the docker container? I'm running it on a Synology NAS. 

If I run paperless-ng locally in a dev environment, then the behavior is correct, i.e. the UI gets updated accordingly.

**Comment by Fogest** [comment]
I am running it on a Ubuntu 20.04 machine and using docker as well.

I'm just wondering if my ssl reverse proxy has anything to do with the
problem. Though my old installation was reverse proxied as well and was
working.

**Comment by siancu** [comment]
Actually in my case the reverse proxy was breaking it. If I go directly to http://localname.homelab:8100 then it works.

**Comment by Fogest** [comment]
Interesting, I'll have to check my reverse proxy setup when I get home and
compare it to my setup on my old machine. My old machine was reverse
proxied as well and didn't have this issue. Though I feel like I recall
having a similar problem in the beginning and I can't remember what fixed
it. I'll have to see if Apache is throwing any errors or if any errors are
sitting in the browser console when I upload.
